src/basic/ordered-set.h

```diff
@@ -62,6 +62,16 @@ static inline char **ordered_set_get_strv(OrderedSet *s) {
         return _hashmap_get_strv(HASHMAP_BASE((OrderedHashmap*) s));
 }
 
+static inline int ordered_set_ensure_put(OrderedSet **s, const struct hash_ops *ops, void *p HASHMAP_DEBUG_PARAMS) {
```

Comment:
Please see how set_ensure_put is defined. This needs to be done the same. (It was better in one of the previous versions of your patch. I don't know why you changed it back.)